packages/opencensus-core/src/trace/model/no-record/no-record-span-base.ts

```diff
@@ -34,7 +34,7 @@ export abstract class NoRecordSpanBase implements types.Span {
   /** The Span ID of this span */
   readonly id: string;
   /** An object to log information to */
-  logger: Logger;
+  logger!: Logger;
```

Comment:
Maybe default this to a console logger or even a `NopLogger` that does nothing? That would prevent exceptions if people try to invoke log functions on a NoRecordSpan I would think.